subsys/bluetooth/host/gatt.c

```diff
@@ -852,6 +904,8 @@ static void db_hash_process(struct k_work *work)
 		 */
 		sc_indicate(0x0001, 0xffff);
 
+		set_all_change_unaware();
```

Comment:
There is an edge case I think. It could appear right after the DFU.

We see that database hash does not match, so we store new hash and set change awareness to false (keep in mind the change awareness store operation is delayed). At this point we may be rebooted and after the reboot we no longer see that the database hash was modified (so we forgotten that peers should have been marked as change unaware).

Should not have high impact anyway since we would most probably send SC indications.

Reply:
There is an edge case I think. It could appear right after the DFU.

We see that database hash does not match, so we store new hash and set change awareness to false (keep in mind the change awareness store operation is delayed). At this point we may be rebooted and after the reboot we no longer see that the database hash was modified (so we forgotten that peers should have been marked as change unaware).

Should not have high impact anyway since we would most probably send SC indications.

Reply:
hm that is true, I did misunderstand you. 
The best way would be to store a "last seen hash" for each peer. I would do that if storage was free. 
I've changed it now to store to settings in `set_all_change_unaware`.

Reply:
Should be good enough.
I think it's better to force a few unnecessary discoveries than to miss a necessary discovery.